controllers/csiscaleoperator_controller.go

```diff
@@ -324,40 +338,48 @@ func (r *CSIScaleOperatorReconciler) getAccessorAndFinalizerName() (metav1.Objec
 	return accessor, finalizerName, nil
 }
 
-func (r *CSIScaleOperatorReconciler) deleteClusterRolesAndBindings() error {
-	/*	if err := r.deleteClusterRoleBindings(instance); err != nil {
+func (r *CSIScaleOperatorReconciler) deleteClusterRolesAndBindings(instance *csiscaleoperator.CSIScaleOperator) error {
+	logger := csiLog.WithName("in deleteClusterRolesAndBindings")
+
+	logger.Info("calling deleteClusterRoleBindings")
+	if err := r.deleteClusterRoleBindings(instance); err != nil {
 		return err
-	}*/
+	}
 
-	if err := r.deleteClusterRoles(); err != nil {
+	logger.Info("calling deleteClusterRoles")
+	if err := r.deleteClusterRoles(instance); err != nil {
 		return err
 	}
 	return nil
 }
 
-func (r *CSIScaleOperatorReconciler) deleteClusterRoles() error {
+func (r *CSIScaleOperatorReconciler) deleteClusterRoles(instance *csiscaleoperator.CSIScaleOperator) error {
+	logger := csiLog.WithName("in deleteClusterRoles")
 
-	/*	clusterRoles := r.getClusterRoles(instance)
+	logger.Info("entering deleteClusterRoles")
+	clusterRoles := r.getClusterRoles(instance)
 
-		for _, cr := range clusterRoles {
-			found := &rbacv1.ClusterRole{}
-			err := r.Client.Get(context.TODO(), types.NamespacedName{
-				Name:      cr.Name,
-				Namespace: cr.Namespace,
-			}, found)
-			if err != nil && errors.IsNotFound(err) {
-				continue
-			} else if err != nil {
-				logger.Error(err, "failed to get ClusterRole", "Name", cr.GetName())
+	for _, cr := range clusterRoles {
+		found := &rbacv1.ClusterRole{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+		}, found)
+		if err != nil && errors.IsNotFound(err) {
+			logger.Info("continuing deleteClusterRoles case err != nil && errors.IsNotFound(err)")
```

Comment:
**Suggestion:** If this log is only for developers then we may comment it. Operator should log more generic messages in logs for cluster admins to understand. However we should also make sure logs have enough data.

Reply:
Done.

Reply:
Thanks for fixing this, can you do the same changes in these methods too `deleteClusterRoleBindings`  `deleteCSIDriver`